src/panels/ActionsPanel/CharmActionsPanel/CharmActionsPanel.tsx

```diff
@@ -210,7 +122,18 @@ export default function CharmActionsPanel({
             />
           </RadioInputBox>
         ))}
-        {generateConfirmationModal()}
+        <ConfirmationDialog
+          confirmType={confirmType}
+          selectedAction={selectedAction}
+          selectedApplications={selectedApplications}
+          setConfirmType={setConfirmType}
+          selectedActionOptionValue={
+            // {} is a placeholder value, as ConfirmationDialog will not render
+            // if selectedAction is undefined.
```

Comment:
I think if we're needing to do this to get around typing issues, then it's probably a sign that we need to move the `selectedAction` check into this component (`{selectedAction ? <ConfirmationDialog ... /> : null}`).